src/python/ensembl/io/genomio/gff3/id_allocator.py

```diff
@@ -0,0 +1,191 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Check and allocate IDs for gene features in a GFF3 file."""
+
+__all__ = ["StableIDAllocator", "InvalidStableID"]
+
+from dataclasses import dataclass, field
+import logging
+import re
+from typing import List, Set
+
+from Bio.SeqFeature import SeqFeature
+
+
+class InvalidStableID(ValueError):
+    """Raised when there is a problem with an ID."""
+
+
+@dataclass
+class StableIDAllocator:
+    """Set of tools to check and allocate stable IDs."""
+
+    # Multiple parameters to automate various fixes
+    skip_gene_id_validation: bool = False
+    min_id_length: int = 7
+    current_id_number: int = 0
+    make_missing_stable_ids: bool = True
+    prefix: str = "TMP_"
+    _loaded_ids: Set = field(default_factory=set)
```

Comment:
Quite a neat trick! :nerd_face:

Reply:
Yeah, this is to avoid setting the default as just `set()` which has unintended consequences if we use more than 1 instance of the class